### PR TITLE
feat: auto-merge based on multiple checks with status comment

### PR DESCRIPTION
**Pull request by anbraten**
Screenshot from 2021-05-22 09-56-11

# TODO
- [ ] ~~parse current comment and get status out of it~~ for now we just "spam" gitlab
- [x] separate merge checks in individual files
- [x] add plugin readme

# Checks
- [ ] specific reviewers approved if a specific label has been applied
- [ ] a merge-request template has been used
- [x] pipeline passes
- [x] no open conversations
- [x] someone approved
- [x] has needed labels
- [x] has no file conflicts
- [x] is not work-in-progress

### CODE REVIEW

File: plugins/auto_merge/merge_status.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	}
+}
+
+func encodeMergeStatus(status *mergeStatus) string {
+	comment := authorTag + "\n"
+	comment = comment + "### Your current merge request status is:\n\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.hasConflicts) + " Your changes do not have conflicts with the target branch\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.openDicussions) + " All discussions about your changes have been resolved\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.passingPipeline) + " A pipline successfully tested your code\n"
+	comment = comment + "- " + getMergeStatusLevelEmojis(status.enoughApprovals) + " Enough reviewers liked your code\n"
```
Comment (lukashass) [inline plugins/auto_merge/merge_status.go:46]: I think these messages are kind of misleading. I would propose a simple description like `conflicts with the target branch` or `discussions resolved` letting the icon be the only status indicator.

Another possibility would be changing messages when the status changes:

- :poop: Your changes are conflicting with the target branch
- :green_heart: Your changes do not have conflicts with the target branch

Comment (anbraten) [reply plugins/auto_merge/merge_status.go:47]: @lukashass Sure we should further improve them 🙃. I like your second approach more as I would like to keep it thematic with some Lassie specialties 😉

Comment (anbraten) [reply plugins/auto_merge/merge_status.go]: @lukashass Could you have a look at the latest changes. I added some changing status messages.

Comment (lukashass) [reply plugins/auto_merge/merge_status.go]: @anbraten Pretty cool stuff :+1:

**Review by lukashass — APPROVED** [review]

**Pull request merged**